vsphere/resource_vsphere_file.go

```diff
@@ -70,6 +70,7 @@ func resourceVSphereFile() *schema.Resource {
 			"create_directories": {
 				Type:     schema.TypeBool,
 				Optional: true,
+				Default:  true,
```

Comment:
Unfortunately for boolean fields that are optional, they have an implicit default of false. The logic DOES check `d.GetOk` where `ok` is supposed to
>...returns the data for the given key and whether or not the key has been set to a non-zero value at some point.

But I honestly don't know if that's true with booleans, they have had a troubled history.

If `GetOk` did behave the way it's described, then in theory this wouldn't be a breaking change, BUT then it would be a bug since we would be ignoring the now new default of `true`.

The logic would need to switch from
```
if v, ok := d.GetOk("create_directories"); ok {
		f.createDirectories = v.(bool)
	}
```
to
```
f.createDirectories = d.Get("create_directories").(bool)
```

I'm boing to recommend we avoid changing the default in this PR and follow up later

Reply:
That's what I thought. Let me update it to remove the default setting now.

Reply:
I've removed the breaking changes.

Reply:
Resubmitted for review.